core/src/main/scala/spinal/core/internals/ComponentEmitterVerilog.scala

```diff
@@ -135,12 +150,26 @@ class ComponentEmitterVerilog(
         //        expressionToWrap ++= mux.inputs
       }
     }
-
+    
+    
     component.children.foreach(sub => sub.getAllIo.foreach(io => if(io.isOutput) {
       val name = component.localNamingScope.allocateName(sub.getNameElseThrow + "_" + io.getNameElseThrow)
-      declarations ++= emitExpressionWrap(io, name)
+      //declarations ++= emitExpressionWrap(io, name)
       referencesOverrides(io) = name
+      
+      var addDecl = true
+      if(spinalConfig.removeOutputAssigns){
+        component.getAllIo.foreach(topio => if(topio.isOutput) {
+          if(topio.head.source == io){
```

Comment:
Which case you try to avoid here ?
if(topio.head.source == io) is weird, it wil likely fail in case of multiple assignement

Reply:
Hmm, are you saying in the case of multiple assignment on the `topio` in this case? I would need to try that out, maybe I didn't try that in my testing. Let me give that a shot and I can modify if needed.

I basically want to remove the intermediate wire/assign that is produced for subcomponent outputs if they connect to another output port of the component they are instantiated in. If they do not connect to an output port (used internally) then I'm ok with the intermediate wire.

Reply:
Hey @Dolu1990 ,

Can you give me an example of what you think would break this? I've tried a few different things but haven't been able to break the proposed fix.

I tried to assign the same output multiple times, but I get an ASSIGNMENT OVERLAP compiler error.

Reply:
Sorry for the delay.

Basicaly : 

```scala
val a = out Bool()
a := subComponent.x
when(somthing){
  a := True
}
```

In such case, the code will enter the  if(topio.head.source == io){  but the issue is that's not the only assignement.

Reply:
To check if a given signal has only one assignement, then : 
```scala
topio.dlcHasOnlyOne
which is equivalent to 
dlcHead == dlcLast && dlcHead != null
```

Reply:
But i'm still worrying about the delta cycle / race conditions issues that such update can create.

Do you see why ?

Reply:
Yes you are correct on the multiple assignment. I tried out the dlcHasOnlyOne check and that seemed to resolve the issue for the few cases I tried.

For the delta cycle/race issue I can see where this could possibly cause some of that to happen. If I can, I'd like to describe the motivation behind removing the assigns. This is from a purely ASIC flow perspective with a heavy mixed signal influence.

When synthesizing with Design Compiler or Genus the tool will often times put in a DRC (design rule check) buffer when an assign is seen. This normally happens when synthesizing with a .lib for an analog component or some other digital PnR block that has been run through PD. For digital logic this is normally ok, as you don't really care about additional buffers provided logical equivalence and timing are preserved. For an analog signal such as a bias, high voltage signal, power/ground, etc. this would obviously be an issue. Since Verilog/VHDL have no way to distinguish digital signals vs. some analog signal that is handled by layout, you have to constrain the design during synthesis to "dont_touch" certain signals.

For example, say you have the following verilog

```verilog
module mod_with_some_analog(
  input  wire some_digital_sig1,
  input  wire some_digital_sig2,
  output wire some_digital_sig3,
  output wire dac, //analog
  input  wire vcc,  //analog
  input  wire vss //analog
);
//...
endmodule
```

`dac`, `vcc`, and `vss` are analog signals in which you would not want the synthesis tool to place a buffer on. To resolve this, you would have the following constraints

```tcl
set_dont_touch [get_pins path_to_mod/dac]
set_dont_touch [get_pins path_to_mod/vcc]
set_dont_touch [get_pins path_to_mod/vss]
```

When an additional assign is put in there is now the *potential* issue that you have two signals to set a dont_touch on. Now I will admit I have not done enough testing to know if the `set_dont_touch` propagates through `assign` statements. It's entirely possible that it does, but I know in the past it has been pushed that we try to remove assign statements for signals which are done in custom layout.

For additional race conditions I'll say that in a traditional ASIC flow we carefully construct the clock trees in such a way that should prevent, or at least significantly reduce the possibility of a delta cycle issue. To put into perspective, in the past 7 years I've only seen it with our analog models where we are doing some rather strange modeling :)

For all of the clocks we use in a design, we actually hand place clock buffers (based on the process/vendor) which are used during synthesis and PnR as a anchor for any design constraints. This is done for a few reasons:
1. Place and Route can modify the respective clock buffers based on speed, power requirements, etc.
2. For design constraints you have a clear point of clock definition. Most synthesis tools will also rename flops in the design. Take the following image
![image](https://user-images.githubusercontent.com/52458420/74590987-c8018480-4fe1-11ea-83cd-b4356ecd9571.png)

Here would be a traditional clock divider where I'm taking the input clock `clk` and creating a divide by 2. One would think to create a generated clock on the `Q` of `FF1`. 
`create_generated_clock -name CLK_DV2 -source [get_ports clk] -edges {1 3 5} [get_pins FF1/Q]`
However during synthesis, `FF1` may be renamed to something like `FF1_reg` or in the case of using multi-bit flops `FF1_SomeOtherReg_MBit`. Sometimes the synthesis tool will print out an updated SDC file with any changes, but this can often cause issues when you merge constraints between IPs.

So we place a clock buffer, that we have set as `dont_touch` so the synthesis/PnR tool will not change it, and we define the clock on that.
`create_generated_clock -name CLK_DV2 -source [get_ports clk] -edges {1 3 5} [get_pins CBUF/Z]`

We now know that this constraint can be used at any stage of the design, pure verilog, Synthesis netlist, PnR netlist for GLS and don't have to worry about constraints referencing a component that is not there.

We also will place a clock mux at the entry point to any IP to allow muxing of a scan clock for DFT purposes. So all flops in a design would be traced back to one conceptual entry point.

That's a long winded answer :) but I also decided that it would be best to have a spinalConfig option to enable/disable this feature, since in the issue you had mentioned the same problem regarding delta cycle issues. 

Are you ok with this feature provided we keep the original methodology (output assigns) as the default?

Reply:
Also I'll do some more testing and update the pull request. I might need a few days with other things at work.